### DISCUSSION TITLE
hi, does iTwin.js now support element motion animation?

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#372] [Q&A] [answered]
hi, does iTwin.js now support element motion animation? For example, there is a door in the model, and I want to achieve a dynamic effect of opening or closing the door.

**Comment by pmconne** [comment, marked as answer]
A [DisplayStyle](https://github.com/imodeljs/imodeljs/blob/c033f09315c7e1ce61ba48fe16c4f33eb7c28330/core/frontend/src/DisplayStyleState.ts#L169) can have an associated [schedule script](https://github.com/imodeljs/imodeljs/blob/c033f09315c7e1ce61ba48fe16c4f33eb7c28330/core/common/src/RenderSchedule.ts#L12). The script can apply rigid body transforms to groups of elements along a timeline. I think @Ellord207 has plans to add an example to our [sample showcase](https://www.itwinjs.org/sample-showcase/) in the near future.
However, this feature is designed to animate the model based on its construction timeline - not for dynamically animating individual elements.
If you are feeling adventurous, you could try writing a [Decorator](https://www.itwinjs.org/learning/frontend/viewdecorations/?term=decorator) that uses the very new [requestElementGraphics](https://github.com/imodeljs/imodeljs/blob/c033f09315c7e1ce61ba48fe16c4f33eb7c28330/core/common/src/rpc/IModelTileRpcInterface.ts#L96) API to obtain graphics for your door and display them with the appropriate transform while hiding the original door graphics using [symbology overrides](https://www.itwinjs.org/reference/imodeljs-frontend/rendering/featuresymbology/featuresymbology.overrides/?term=featuresy)

**Reply by shenshengyi** [reply]
OK, thank you very much for your answer. I'll try it with requestelement graphics and decorator as you suggest.
